### PR TITLE
Update CONTRIBUTING and README

### PR DESCRIPTION
**Pull request by emilmelnikov**
Make README more appealing, bring the contributing instructions up
to date with the current available tools, and add more copy-pasteable
command snippets.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (k-dominik) [inline CONTRIBUTING.md:24]: I'm not buying into mamba just yet. Our whole distribution stack still uses conda and in order to get the same environments, I'd still prefer to use conda. See also https://github.com/mamba-org/mamba/issues/525

Comment (k-dominik) [reply CONTRIBUTING.md:24]: for `ilastik-dependencies-no-solvers` we'd get the following diff:

<details>

```
Common Packages (with diff)
---------------------------
  importlib-metadata
    build_number: 0 -> 1
    build_string: py37hc8dfbb8_0 -> py_1
    dist_name: importlib-metadata-2.0.0-py37hc8dfbb8_0 -> importlib-metadata-2.0.0-py_1
    platform: linux-64 -> noarch
  jsonschema
    build_number: 1 -> 2
    build_string: py37hc8dfbb8_1 -> py_2
    dist_name: jsonschema-3.2.0-py37hc8dfbb8_1 -> jsonschema-3.2.0-py_2
    platform: linux-64 -> noarch
  libblas
    build_number: 1 -> 17
    build_string: 1_h86c2bf4_netlib -> 17_openblas
    dist_name: libblas-3.9.0-1_h86c2bf4_netlib -> libblas-3.8.0-17_openblas
    version: 3.9.0 -> 3.8.0
  libcblas
    build_number: 1 -> 17
    build_string: 1_h86c2bf4_netlib -> 17_openblas
    dist_name: libcblas-3.9.0-1_h86c2bf4_netlib -> libcblas-3.8.0-17_openblas
    version: 3.9.0 -> 3.8.0
  liblapack
    build_number: 1 -> 17
    build_string: 1_h86c2bf4_netlib -> 17_openblas
    dist_name: liblapack-3.9.0-1_h86c2bf4_netlib -> liblapack-3.8.0-17_openblas
    version: 3.9.0 -> 3.8.0
  libpq
    build_string: h1281834_2 -> h5513abc_2
    dist_name: libpq-12.3-h1281834_2 -> libpq-12.3-h5513abc_2
  nose
    build_number: 1004 -> 1006
    build_string: py37hc8dfbb8_1004 -> py_1006
    dist_name: nose-1.3.7-py37hc8dfbb8_1004 -> nose-1.3.7-py_1006
    platform: linux-64 -> noarch
  pyopenssl
    build_number: 0 -> 1
    build_string: py37_0 -> py_1
    dist_name: pyopenssl-19.1.0-py37_0 -> pyopenssl-19.1.0-py_1
    platform: linux-64 -> noarch
  python
    build_string: h425cb1d_1_cpython -> h6f2ec95_1_cpython
    dist_name: python-3.7.8-h425cb1d_1_cpython -> python-3.7.8-h6f2ec95_1_cpython
  scikit-learn
    build_number: 0 -> 1
    build_string: py37_blas_openblashebff5e3_0 -> py37ha8026db_1
    dist_name: scikit-learn-0.20.4-py37_blas_openblashebff5e3_0 -> scikit-learn-0.20.3-py37ha8026db_1
    version: 0.20.4 -> 0.20.3

Packages Only In /tmp/mamba-env.json:
--------------------------
  blas
  libgfortran
  openblas

Packages Only In /tmp/conda-env.json:
--------------------------
  libopenblas
```

</details>

Comment (emilmelnikov) [reply CONTRIBUTING.md:24]: Thanks for pointing that, I considered mamba to be a drop-in replacement before.

**Review by k-dominik — COMMENTED** [review]
I think this is fine in general. Before merging I would be interested who in the team is actually using gh-cli (I'm not). Hence I would've written the guidelines slightly differently. Maybe a good one to discuss in the meeting!

**Comment by emilmelnikov** [comment]
> Before merging I would be interested who in the team is actually using gh-cli (I'm not).

This is mainly for newcomers who want to quickly set up the environment, you don't really need gh-cli for those steps, it's just easier to tell people to run 2 commands instead of "go to this web page, fork the repo, clone the repo locally, add upstream remotes etc.". gh-cli also authenticates you by opening a browser.

File: CONTRIBUTING.md
Comment (k-dominik) [reply CONTRIBUTING.md:30]: okay, conda develop is not a package.

Comment (k-dominik) [reply CONTRIBUTING.md:30]:
```suggestion
1. Install [conda-build](https://docs.conda.io/projects/conda-build/en/latest/) in order to access the [conda develop command](https://docs.conda.io/projects/conda-build/en/latest/resources/commands/conda-develop.html):
```

File: CONTRIBUTING.md
Comment (k-dominik) [inline CONTRIBUTING.md:32]:
```suggestion
    conda install --name base conda-build
```

File: CONTRIBUTING.md
Comment (k-dominik) [inline CONTRIBUTING.md:77]: might be a little overkill to use `-C` flag for git. I always `cd` into directories... I mean this is directed at novices and you're adding a flag to git without explanation.

File: CONTRIBUTING.md
Comment (k-dominik) [inline CONTRIBUTING.md:83]: this seems to imply that you _always_ checkout a branch on both, which is seldom the case.

**Review by k-dominik — COMMENTED** [review]
Ah I really like the new readme, it sounds so friendly!

One thing that is missing is the hint on text-only changes that can also be done via the github web ui.

**Comment by emilmelnikov** [comment]
> One thing that is missing is the hint on text-only changes that can also be done via the github web ui.

This is actually one more argument for moving our to RTD: they have the "Edit on GitHub" button.

**Review by k-dominik — APPROVED** [review]
very nice and concise :)

I'm still not fully sure wrt gh-cli, but willing to go with the flow.

**Comment by k-dominik** [comment]
any reason this is not in yet @emilmelnikov ?

**Comment by emilmelnikov** [comment]
> any reason this is not in yet @emilmelnikov ?

I thought we wanted to add something about creating a conda package from sources.

**Comment by k-dominik** [comment]
this should not hold this one back. I've added a todo for myself https://github.com/ilastik/ilastik/issues/2364 ...

**Pull request merged**
